packages/devtools_app/lib/src/screens/inspector_v2/inspector_controller.dart

```diff
@@ -462,6 +462,7 @@ class InspectorController extends DisposableController
       }
       if ((_receivedFlutterNavigationEvent || _receivedIsolateReloadEvent) &&
           extensionEventKind == 'Flutter.Frame') {
+        _refreshingAfterNavigationEvent = _receivedFlutterNavigationEvent;
```

Comment:
could there be a race where this is set again (after receiving another event here) before we reach the if statement you added below?

Reply:
Yes, it could. I don't think it's likely and the consequence would just be we would then walk down the tree to find the closest matching, so it's not the worst outcome. However, I've added a TODO (below). I initially was passing the value for "refreshingAfterNavigationEvent" down the call chain, but realized that would involve changing the interface to `onForceRefresh` for the InspectorServiceClient which I didn't want to do: https://github.com/flutter/devtools/blob/65ceb669b37eb3758a3f2874240dfe8d4ab81618/packages/devtools_app/lib/src/shared/diagnostics/inspector_service.dart#L1212